.github/workflows/dotnet.yml

```diff
@@ -48,3 +49,43 @@ jobs:
           export DISPLAY=:1.5
           cd lib/PuppeteerSharp.Tests
           dotnet test -f net6.0 -s test.runsettings -c Debug --logger "trx;LogFileName=TestResults.xml"
+
+  deployRelease:
+    if: startsWith( github.ref, 'refs/heads/release-' )
+    runs-on: ubuntu-latest
+    needs: build
+    steps:
+    - uses: actions/checkout@v3
+    - name: Setup .NET
+      uses: actions/setup-dotnet@v3
+      with:
+        dotnet-version: 6.0.x
+    - name: Build
+      run: dotnet build  lib/PuppeteerSharp.sln --configuration Release --no-restore
+    - name: NugetPush to nuget.org
+      env:
+        NUGET_TOKEN_EXISTS: ${{ secrets.NUGET_TOKEN  }}
+      if: env.NUGET_TOKEN_EXISTS != '' && startsWith( github.ref, 'refs/heads/release-' )
+      run: |
+        ls ./lib/PuppeteerSharp/bin/Release
+        dotnet nuget push ./lib/PuppeteerSharp/bin/Release/*.nupkg --skip-duplicate --api-key ${{secrets.NUGET_TOKEN}} --source https://api.nuget.org/v3/index.json
+
+  deployMaster:
+    if: github.ref == 'refs/heads/master'
```

Comment:
MS recommends it. It will apply validation before going to do a release build...  your the maintainer.... removing.
FYI
_In addition, uploading packages to [int.nugettest.org](https://int.nugettest.org/) is a great way to test and validate your package without making a package available in the public gallery. API Endpoint: https://apiint.nugettest.org/v3/index.json_
https://learn.microsoft.com/en-us/nuget/nuget-org/nuget-org-faq

Reply:
Ahh that's a good one. Let's do it. I'm creating a secret for that.

Reply:
Secret `NUGET_INT_API_KEY` created.